### DISCUSSION TITLE
get Texture as blob/base64/img

### DISCUSSION THREAD

**Discussion by shamsail** [google/model-viewer#2531] [Q&A]
Hi,
I want to get texture as blob from viewer. I know that we can set a new texture source to a specific material by this command.

`material.occlusionTexture.texture.source.setURI("path/to/texture.png");
`
and it's working, like to change texture on a material.

But how can I access texture as blob/base64/img ? (on load)

Because it's not documented anywhere.

I can access 

`material.occlusionTexture.texture
`
and
 `material.occlusionTexture.texture.source
`
But how I can make manipulation to get texture and show as image

**Comment by shamsail** [comment]
Hi, I have found a dirty solution for this so that I can get texture as image.
I'm doing some other manipulation to achieve this. Like to get symbol first from the `material.occlusionTexture.texture.source` like this

```
 
let source = material.occlusionTexture.texture.source;
let sym = Object.getOwnPropertySymbols(source).find(function (s) {
      return String(s) === "Symbol(correlatedObjects)";
});


```

Now, after getting the symbol, I'm getting image out of it like this

```
source[sym].entries().next().value[0].image
```

I'm using this image to show on DOM like this.

```
let ImageDOM = new Image(100, 100);
ImageDOM = source[sym].entries().next().value[0].image;
document.body.appendChild(ImageDOM);
``` 

This is a dirty solution what I'm using right now but it's working fine. 
There should be some property or get method to do this.

Setting texture image is quite nice but there is no simple way to get the texture image. Dated:1st July,2020

**Reply by shamsail** [reply]
waiting for a better solution in which we can get direct texture image @elalish

**Reply by shamsail** [reply]
It's not an ideal solution.. Ideal solution should be like this.. 
```
material.occlusionTexture.texture.source.get()
```
or

```
material.occlusionTexture.texture.source.getImage()
```

There is `.uri` method, and I don't know what it is? because it's undefined. `.setURI` is working fine. main issue is to give a proper access to the texture

**Reply by shamsail** [reply]
may be I'm wrong and you guys have implemented such thing to do this stuff. but the thing is , it's not documented (if there is some easy solution)

**Reply by elalish** [reply]
Believe it or not, I just implemented the feature you describe because I needed it for our editor. I haven't merged it yet, but it should come out with our next release.

**Reply by shamsail** [reply]
perfect. because it's hectic to do like this, what's I'm currently doing. (going into further inner detail) . Inner implementation in library will give better approach. How we can access this feature in next release?

**Comment by base-krchome** [comment]
#2562
